### ISSUE TITLE
Empty user-agent allows unauthenticated access

### ISSUE THREAD

**Issue by jowi24** [oauth2-proxy/oauth2-proxy#657]
## Expected Behavior

As with version 5.1.1, a simple curl request to a oauth2 protected resource results in status code 401 as no cookie is presented. The oauth proxy log shows:
```
[2020/07/04 13:26:44] [oauthproxy.go:871] Error loading cookied session: Cookie "_oauth2_proxy" not present
172.18.0.xxx - - [2020/07/04 13:26:44] xxx GET - "/oauth2/auth" HTTP/1.1 "" 401 
```

## Current Behavior

As with version 6.0. the curl request with no user-agent set (-A "") results in, even with no auth cookie presented.
```
172.18.0.xxx - - [2020/07/04 13:25:22] xxx GET - "/oauth2/auth" HTTP/1.1 "" 200 
```

## Steps to Reproduce (for bugs)

1.  `curl -A "" https://server/protected-resource`

## Your Environment

- Version used: 6.0.0

**Comment by JoelSpeed** [comment]
This is a duplicate of #649 and will be fixed by #650, I'm just waiting for the PR author to fix it up and then I'll get it merged 

The empty user agent will only display ok though so for direct access you're ok. If you are using the OAuth2 Proxy as an auth request module, you may be able to add a workaround to the proxy to set a user agent if it is empty

**Comment by jowi24** [comment]
Thank you! Did not recognize that #649 discusses the same issue. I'll try the suggested workaound 👍
